plugin/src/main/java/org/opensearch/ml/processor/ModelExecutor.java

```diff
@@ -130,11 +177,35 @@ default Object getModelOutputValue(ModelTensorOutput modelTensorOutput, String m
                 throw new RuntimeException("Model outputs are null or empty.");
             }
         } catch (Exception e) {
-            throw new RuntimeException("An unexpected error occurred: " + e.getMessage());
+            throw new RuntimeException(e.getMessage());
         }
         return modelOutputValue;
     }
 
+    default Object getModelOutputValue(MLOutput mlOutput, String modelOutputFieldName, boolean ignoreMissing, boolean fullResponsePath) {
+        try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
+            String modelOutputJsonStr = mlOutput.toXContent(builder, ToXContent.EMPTY_PARAMS).toString();
+            Map<String, Object> modelTensorOutputMap = gson.fromJson(modelOutputJsonStr, Map.class);
+            if (!fullResponsePath && mlOutput instanceof ModelTensorOutput) {
+                return getModelOutputValue((ModelTensorOutput) mlOutput, modelOutputFieldName, ignoreMissing);
+            } else if (modelOutputFieldName == null || modelTensorOutputMap == null) {
+                return modelTensorOutputMap;
+            } else {
+                try {
+                    return JsonPath.parse(modelTensorOutputMap).read(modelOutputFieldName);
+                } catch (Exception e) {
+                    if (ignoreMissing) {
+                        return modelTensorOutputMap;
+                    } else {
+                        throw new IllegalArgumentException("model inference output cannot find such json path: " + modelOutputFieldName, e);
+                    }
+                }
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("An unexpected error occurred: " + e.getMessage());
```

Comment:
Shall we also add `ignoreMissing` checking here? If true, return a result?

Reply:
good catch. But we are adding it [here](https://github.com/rbhavna/ml-commons/blob/ml_ingest_processor/plugin/src/main/java/org/opensearch/ml/processor/ModelExecutor.java#L262) that will check the ignoreMissing when we call `getModelOutputValue` in line line 190. It should probably cover all the other cases.